api/src/extensions/lib/get-extensions.ts

```diff
@@ -8,7 +8,15 @@ export const getExtensions = async () => {
 
 	const loadedNames = localExtensions.map(({ name }) => name);
 
-	const filterDuplicates = ({ name }: Extension) => loadedNames.includes(name) === false;
+	const filterDuplicates = ({ name }: Extension) => {
+		const isUnique = loadedNames.includes(name) === false;
+
+		if (isUnique) {
+			loadedNames.push(name);
+		}
+
+		return isUnique;
+	};
 
 	const localPackageExtensions = (await resolvePackageExtensions(getExtensionsPath())).filter((extension) =>
 		filterDuplicates(extension),
```

Comment:
I don't like `.filter` being used with side effects, I'd prefer a `for` loop here which pushes to `localPackageExtensions`, in combination with a `Set` (`loadedNames.has`) for unique checks.

Reply:
Absolutely agree `filter`/`map`/`sort` should never have side effects.